dbms/src/Storages/S3/S3Common.cpp

```diff
@@ -596,10 +604,20 @@ static bool doUploadFile(
     Aws::S3::Model::PutObjectRequest req;
     client.setBucketAndKeyWithRoot(req, remote_fname);
     req.SetContentType("binary/octet-stream");
-    auto istr
-        = Aws::MakeShared<Aws::FStream>("PutObjectInputStream", local_fname, std::ios_base::in | std::ios_base::binary);
-    RUNTIME_CHECK_MSG(istr->is_open(), "Open {} fail: {}", local_fname, strerror(errno));
     auto write_bytes = std::filesystem::file_size(local_fname);
+    RandomAccessFilePtr local_file;
+    if (file_provider)
+    {
+        local_file = file_provider->newRandomAccessFile(local_fname, encryption_path, nullptr);
+    }
+    else
+    {
+        local_file = PosixRandomAccessFile::create(local_fname);
+    }
+    // Read at most 16MB each time.
+    auto read_buf = std::make_unique<ReadBufferFromRandomAccessFile>(local_file, 16 * 1024 * 1024);
+    auto istr = std::make_shared<StdStreamFromReadBuffer>(std::move(read_buf), write_bytes);
```

Comment:
```suggestion
    auto istr = Aws::MakeShared<StdStreamFromReadBuffer>("PutObjectInputStream", std::move(read_buf), write_bytes);
```

Reply:
The `istr` finally will be released in `PutObjectRequest`. Use `Aws::MakeShared` for it or ASAN may throw warning message like https://github.com/pingcap/tiflash/pull/8618

Reply:
done